integration/spark-common/src/main/scala/org/apache/spark/sql/catalyst/CarbonDDLSqlParser.scala

```diff
@@ -400,6 +403,39 @@ abstract class CarbonDDLSqlParser extends AbstractCarbonSparkSQLParser {
       tableComment)
   }
 
+  /**
+   * this method validates whether any of the child columns of complex dataType column
+   * is of type string.
+   */
+  def validateChildColumns(field: Field): Boolean = {
```

Comment:
can you change the logic to return true as soon as you find any column with string datatype. No need to check other columns.